liquigraph-core/src/main/java/org/liquigraph/core/configuration/validators/UserCredentialsOptionValidator.java

```diff
@@ -32,8 +33,23 @@
     public Collection<String> validate(String username, String password) {
         if (username != null ^ password != null) {
             return Collections.singletonList("Please provide both username and password, or none.");
-        } else {
+        }
+        return Collections.emptyList();
+    }
+
+
+    /**
+     * Validates the given user credentials are provided; username and password
+     *
+     * @param username the username to use to connect
+     * @param password the password corresponding to {@code username}
+     * @return a collection of Strings describing possible errors, an empty
+     * collection if no errors
+     */
+    public Collection<String> validateWithDatabase(Optional<String> username, Optional<String> password) {
```

Comment:
`validateWithDatabase` does not seem to include a database parameter

Reply:
maybe could merge the other .validate with it? OR -  enforce providing username and password in the long run, to avoid these check, up to you :)